src/SOS/Strike/util.cpp

```diff
@@ -5721,3 +5721,30 @@ HRESULT GetMetadataMemory(CLRDATA_ADDRESS address, ULONG32 bufferSize, BYTE* buf
 }
 
 #endif // FEATURE_PAL
+
+/**********************************************************************\
+* Routine Description:                                                 *
+*                                                                      *
+*    Since .NET 9+ the runtime does not expose EEClass, but instead    *
+*    returns a pointer to the canonical MethodTable in                 *
+*    DacpMethodTableData:Class.                                        *
+*    Detect that situation by calling GetMethodTableForEEClass and     *
+*    comparing the result to the EEClass itself.                       *
+*                                                                      *
+\**********************************************************************/
+
+HRESULT PreferCanonMTOverEEClassForNET9(CLRDATA_ADDRESS eeClassPtr, BOOL &preferCanonMT, CLRDATA_ADDRESS *outCanonMT)
```

Comment:
nit: this makes callsite a bit clearer (pass-by-ref `&net9preferMT`)
```suggestion
HRESULT PreferCanonMTOverEEClassForNET9(CLRDATA_ADDRESS eeClassPtr, BOOL* preferCanonMT, CLRDATA_ADDRESS *outCanonMT)
```